tests/src/tests/test_utils.go

```diff
@@ -0,0 +1,324 @@
+	} else {
+		if logToFileOnly {
+			testLogger = log.New(f, "", log.LstdFlags|log.Lmicroseconds)
+		} else {
+			writer := io.MultiWriter(f, os.Stdout)
+			testLogger = log.New(writer, "", log.LstdFlags|log.Lmicroseconds)
+		}
+		testLogger.Println("Test Starting:")
+	}
+}
+
+func GetTempDir() string {
+	_, fn, _, ok := runtime.Caller(1)
+	if !ok {
+		return tmpDir
+	}
+	testFile := filepath.Base(fn)
+	testFileExtension := filepath.Ext(testFile)
+	subdir := testFile[0 : len(testFile)-len(testFileExtension)]
+	return CreateTempDir(subdir)
+}
+
+func CreateTempDir(subdir string) string {
+	tempDir := tmpDir + subdir + "/"
+	if _, err := os.Stat(tempDir); os.IsNotExist(err) {
+		os.Mkdir(tempDir, 0755)
+	}
+	return tempDir
+}
+
+func GetDevFileName() string {
+	pc, fn, _, ok := runtime.Caller(1)
+	if !ok {
+		return tmpDir + "DefaultDevfile"
+	}
+
+	testFile := filepath.Base(fn)
+	testFileExtension := filepath.Ext(testFile)
+	subdir := testFile[0 : len(testFile)-len(testFileExtension)]
+	destDir := CreateTempDir(subdir)
+	callerName := runtime.FuncForPC(pc).Name()
+
+	pos1 := strings.LastIndex(callerName, "/tests.") + len("/tests.")
+	devfileName := destDir + callerName[pos1:len(callerName)] + ".yaml"
+
+	LogMessage(fmt.Sprintf("GetDevFileName : %s", devfileName))
+
+	return devfileName
+}
+
+func AddSuffixToFileName(fileName string, suffix string) string {
+	pos1 := strings.LastIndex(fileName, ".yaml")
+	newFileName := fileName[0:pos1] + suffix + ".yaml"
+	LogMessage(fmt.Sprintf("Add suffix %s to fileName %s : %s", suffix, fileName, newFileName))
+	return newFileName
+}
+
+func LogMessage(message string) string {
+	testLogger.Println(message)
+	return message
+}
+
+type TestDevfile struct {
+	SchemaDevFile   schema.Devfile
+	FileName        string
+	ParsedSchemaObj parser.DevfileObj
+	SchemaParsed    bool
+}
+
+var StringCount int = 0
+
+var RndSeed int64 = time.Now().UnixNano()
+
+func GetRandomUniqueString(n int, lower bool) string {
+	StringCount++
+	return fmt.Sprintf("%s%04d", GetRandomString(n, lower), StringCount)
+}
+
+func setRandSeed() {
+	RndSeed++
+	rand.Seed(RndSeed)
+}
+
+const schemaBytes = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz"
+
+func GetRandomString(n int, lower bool) string {
+	setRandSeed()
+	b := make([]byte, n)
+	for i := range b {
+		b[i] = schemaBytes[rand.Intn(len(schemaBytes)-1)]
+	}
+	randomString := string(b)
+	if lower {
+		randomString = strings.ToLower(randomString)
+	}
+	return randomString
+}
+
+func GetGroupKinds() []schema.CommandGroupKind {
+	setRandSeed()
+	return []schema.CommandGroupKind{schema.BuildCommandGroupKind, schema.RunCommandGroupKind, schema.TestCommandGroupKind, schema.DebugCommandGroupKind}
+}
+
+func GetRandomGroupKind() schema.CommandGroupKind {
+	setRandSeed()
+	return GetGroupKinds()[rand.Intn(len(GetGroupKinds()))]
+}
+
+func GetBinaryDecision() bool {
+	return GetRandomDecision(1, 1)
+}
+
+func GetRandomDecision(success int, failure int) bool {
```

Comment:
how does this function works? why expect an int value for success and failure?
Besides, I would like to suggest having a simple comment for every function you added

Reply:
There are a lots of functions without comments. Is it OK if I add these in a subsequent pull request? The tests are working and would like to get them merged - this is only part 1.

Reply:
Have now added simple comments to all functions